tests/extended_temporal_memory/extensive_etm_test.py

```diff
@@ -0,0 +1,1340 @@
+
+  A10: Ineffectiveness of feedback without correct proximal input.
+
+  ============================================================================
+                             Other tests
+  ============================================================================
+
+  O1-O4: Joint learning from proximal/external input with feedback.
+
+  O5-O7: Tests drawn from the previous categories, adding slight spatial noise
+  to input patterns at test time.
+  """
+
+  VERBOSITY = 1
+  n = 2048
+  w = range(38, 43)
+  feedback_size = 400
+
+
+  def testE1(self):
+    """Joint proximal and external first order learning, correct inputs at
+    test time.
+
+    Train on ABCDE/PQRST, test on same sequences should yield perfect
+    predictions.
+    """
+    self.init({"cellsPerColumn": 1})
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    externalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    for _ in xrange(2):
+      self.feedTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self._testTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self.assertAllActiveWerePredicted()
+    self.assertAllInactiveWereUnpredicted()
+
+
+  def testE2(self):
+    """Joint proximal and external first order learning, no external input at
+    test time.
+
+    Train on ABCDE/PQRST, test on ABCDE/None leads to bursting.
+    """
+    self.init({"cellsPerColumn": 1})
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    externalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    for _ in xrange(2):
+      self.feedTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self._testTM(proximalInputA, activeExternalCellsSequence=None)
+    self.assertAllActiveWereUnpredicted()
+
+
+  def testE3(self):
+    """Joint proximal and external first order learning, incorrect external
+    prediction at test time.
+
+    Train on ABCDE/PQRST, test on ABCDE/VWXYZ leads to bursting.
+    """
+    self.init({"cellsPerColumn": 1})
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    externalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    externalInputB = self.sequenceMachine.generateFromNumbers(numbers)
+
+    for _ in xrange(2):
+      self.feedTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self._testTM(proximalInputA, activeExternalCellsSequence=externalInputB)
+    self.assertAlmostAllActiveWereUnpredicted()
```

Comment:
@subutai the randomly generated sequence of external inputs happen to have the same pattern at some index (36 to be precise), and thus the prediction is correct at the next step. I figured this was less of a hack as manually changing this pattern, as we would expect this to happen in real settings as well.